docs/pipeline-configuration-files/specifications.mdx

```diff
@@ -0,0 +1,272 @@
+- **Required**: No
+- **Default**: `{"level": "warn", "message": "record delivery failed"}`
+- **Description**: A map of configuration keys and values for the dead-letter queue connector.
+
+#### window-size
+
+- **Data Type**: Integer
+- **Required**: No
+- **Default**: `1`
+- **Description**: Defines the nack window size. See [dead-letter queue](/docs/features/dead-letter-queue#nack-window-and-threshold).
+
+#### window-nack-threshold
+
+- **Data Type**: Integer
+- **Required**: No
+- **Default**: `0`
+- **Description**: Defines the nack window threshold. See [dead-letter queue](/docs/features/dead-letter-queue#nack-window-and-threshold).
+
+### connectors
+
+- **Data Type**: Sequence node containing [`connector`](#connector) nodes
+- **Required**: No
+- **Default**: None
+- **Description**: The node contains a list of [`connector`](#connector) definitions. The order of the connectors has no
+  effect on the provisioned pipeline.
+
+### processors
+
+- **Data Type**: Sequence node containing [`processor`](#processor) nodes
+- **Required**: No
+- **Default**: None
+- **Description**: The node contains a list of [`processor`](#processor) definitions. These processors process all
+  records flowing through the pipeline. They are executed after source processors and before destination processors. The
+  processors are executed in the order in which they are defined in the list.
+
+## connector
+
+- **Data Type**: Mapping node
+- **Required**: n/a
+- **Default**: None
+- **Description**: This node contains the configuration of a single connector.
+
+### id
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Description**: ID of the [`connector`](#connector). It should uniquely identify the connector inside the pipeline.
+  If multiple connectors inside the pipeline use the same ID, provisioning will fail.
+
+**Warning**: Changing this property will cause the connector to start from the beginning.
+
+### type
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Allowed Values**: `source`, `destination`
+- **Description**: Defines if the connector is a source or a destination. A working pipeline needs at least one source
+  and one destination.
+
+**Warning**: Changing this property will cause the connector to start from the beginning.
+
+### plugin
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Description**: This node references the connector plugin. See how to [reference a connector](/docs/connectors/referencing).
+
+**Warning**: Changing this property will cause the connector to start from the beginning.
+
+### name
+
+- **Data Type**: String
+- **Required**: No
+- **Default**: Same as connector [`id`](#id-1)
+- **Description**: Human readable name for the connector.
+
+### settings
+
+- **Data Type**: Mapping node
+- **Required**: Yes
+- **Default**: None
+- **Description**: A map of configuration keys and values for the connector plugin. The values in this map depend on the
```

Comment:
we could let the user know where to get the needed settings (checking the config table from the connector docs)